### ISSUE TITLE
question: how does DSpot set the number of threads for Pit?

### ISSUE THREAD

**Issue by monperrus** [STAMP-project/dspot#389]
follow-up of a question during the meeting:

- how does DSpot set the number of threads for Pit?
- is this configurable at the Dspot level?

**Comment by danglotb** [comment]
Hi.

> how does DSpot set the number of threads for Pit?

**DSpot** uses **PIT** through its maven plugin, by setting options directly in maven goals, 
_e.g._ `mvn  org.pitest:pitest-maven:1.3.0:mutationCoverage -DwithHistory -DtargetClasses=example.* -DreportsDirectory=target/pit-reports -DoutputFormats=CSV,HTML -DtargetTests=example.TestSuiteExample -DadditionalClasspathElements=target/dspot/dependencies/ -DmutationEngines=gregor -Dmutators=ALL`
So for now, **DSpot** does not set the number of threads for **PIT**.

> is this configurable at the Dspot level?

I guess it could be easily configurable, through CLI or in the properties file.

**Comment by monperrus** [comment]
According to your snippet, it seems that DSpot does not set the number of threads for Pit. Correct?

**Comment by monperrus** [comment]
@jesus-gorronogoitia says that DSpot is too slow in some cases.

@oscarlvp says that this is an important parameter.

Should we set  the number of threads for PIT?

**Comment by jesus-gorronogoitia** [comment]
I don't think it is a problem on the number of threads. DSpot shown to be slow when amplifying tests testing highly intensive I/O invocations of remote services. In particular, minions generating and testing amplified tests were mostly idle (1-5% of CPU consumption) during the entire amplification process. See #386 for execution logs. I also collected the CPU consumption, but as I did it with very high frequency (every 5 sec), this file is quite big (3Gb), so I didn't commit it to the repository. But I can provide you some samples to give you an idea.
The entire test suite takes 50 sec to be executed. In amplification logs I see many I/O access to the remove services, so I guess the generated mutations implies a high number of remote invocations that slow down the amplification process.
